### ISSUE TITLE
[General Question] Could somebody help me figure out my wired training curves

### ISSUE THREAD

**Issue by makecent** [open-mmlab/mmaction2#1780]
My training curve of accuracy is strange:
![Screenshot from 2022-08-12 16-54-59](https://user-images.githubusercontent.com/42603768/184320607-ae7dfccf-12fa-4225-9d4f-84374797543d.png) ![Screenshot from 2022-08-12 17-25-47](https://user-images.githubusercontent.com/42603768/184325923-fbef2d6e-8201-4283-bb8b-8120146d4246.png)

The learning rate is linear warm-up (first epoch) followed by a cosine decaying.
![Screenshot from 2022-08-12 17-19-26](https://user-images.githubusercontent.com/42603768/184324800-2ce3638a-75d6-4f14-8826-afc5ad788979.png)

I think the reason should NOT come from the learning rates because the wired curves **only happen when I run evaluation every half-epoch**:
```python
lr_config = dict(policy='CosineAnnealing',
                 min_lr_ratio=0.01,
                 warmup='linear',
                 warmup_ratio=0.01,
                 warmup_iters=1,
                 warmup_by_epoch=True)
evaluation = dict(by_epoch=False, interval=44640, metrics=['top_k_accuracy'], save_best='top_k_accuracy') 
checkpoint_config = dict(by_epoch=False, interval=44640)
# 44640 is the half number of the iterations of one epoch (89281 iters per epoch)
```
The log:
```shell
# The logs around first half epoch
Epoch [1][44000/89281]	lr: 4.979e-05, loss_reg: 0.4341, reg_acc: 0.7996
Epoch [1][44500/89281]	lr: 5.034e-05, loss_reg: 0.4316, reg_acc: 0.8003
Saving checkpoint at 44640 iterations
Evaluating top_k_accuracy...
Epoch [1][45000/89281]	lr: 5.090e-05, loss_reg: 0.3881, reg_acc: 0.8411

...

# The logs around the first epoch
Epoch [1][88500/89281]	lr: 9.913e-05, loss_reg: 0.2447, reg_acc: 0.9108
Epoch [1][89000/89281]	lr: 9.969e-05, loss_reg: 0.2429, reg_acc: 0.9127
Saving checkpoint at 89280 iterations
Evaluating top_k_accuracy...
Epoch [2][500/89281]	lr: 9.758e-05, loss_reg: 0.2930, reg_acc: 0.8745
```

The training accuracy has an instant rise after the first half-epoch validation, and an instant drop after the first epoch. While the training loss is the opposite.

Do you guys have any ideas about the possible reasons for this? 

When I use a normal evalution config, i.e., evaluation by every epoch, the curves get normal (The green line):
![Screenshot from 2022-08-12 17-35-11](https://user-images.githubusercontent.com/42603768/184327502-f92c6297-26f4-4cce-9853-6421d585515e.png)

**Comment by makecent** [comment]
I figured it out. It's because the state of model.training was switched off.

Specifically, because I evaluate the model at the middile of an epoch, it will turn off the model training state by excuting `model.eval()`.:
https://github.com/open-mmlab/mmaction2/blob/5fa8faa6a7cb2adf9b5eaf9f8a7473b65bb8228d/mmaction/apis/test.py#L76 

And this state `runner.model.training=False` will keep till to the [next epoch](https://github.com/open-mmlab/mmcv/blob/f527e43c1a1e52e5903410255aea1f71f005a161/mmcv/runner/epoch_based_runner.py#L43).

Once the model is in the evaluation state, some regularization method such as the dropout will be deactivated, resulting the training to be easier, thereby lower loss and higher accuracy.
